### DISCUSSION TITLE
Fresh brew uninstall/reinstall on Mac Monterrey 12.7.1 fails in standalone/init.rb with undefined method 'parent' for #<Pathname...

### DISCUSSION THREAD

**Discussion by thn929** [Homebrew/discussions#4931] [Getting started] [answered]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

1. Commands to demonstrate minimum requirements for brew as listed [here](https://docs.brew.sh/Installation)
```shell
~$ sw_vers
ProductName:	macOS
ProductVersion:	12.7.1
BuildVersion:	21G920

~$ uname -m
x86_64

~$ pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | grep version
version: 14.2.0.0.1.1668646533

~$ ps $$
  PID   TT  STAT      TIME COMMAND
11955 s001  S      0:00.08 -bash
```

2. Uninstall and reinstall
```shell
~$ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)"
Warning: This script will remove:
/Users/thn929/Library/Caches/Homebrew/
/usr/local/Cellar/
/usr/local/bin/brew -> /usr/local/bin/brew
Are you sure you want to uninstall Homebrew? This will remove your installed packages! [y/N] y
==> Removing Homebrew installation...
Warning: Failed to delete /Users/thn929/Library/Caches/Homebrew
rm: /Users/thn929/Library/Caches/Homebrew/api: Permission denied
rm: /Users/thn929/Library/Caches/Homebrew: Permission denied
Warning: Failed to delete /usr/local/Cellar
rm: /usr/local/Cellar: Permission denied
==> Removing empty directories...
Password:
==> /usr/bin/sudo /usr/bin/find /usr/local/bin /usr/local/etc /usr/local/include /usr/local/lib /usr/local/opt /usr/local/sbin /usr/local/share /usr/local/var /usr/local/Cellar /usr/local/Homebrew /usr/local/Frameworks -name .DS_Store -delete
==> /usr/bin/sudo /usr/bin/find /usr/local/bin /usr/local/etc /usr/local/include /usr/local/lib /usr/local/opt /usr/local/sbin /usr/local/share /usr/local/var /usr/local/Cellar /usr/local/Homebrew /usr/local/Frameworks -depth -type d -empty -exec rmdir {} ;
==> Homebrew uninstalled!
The following possible Homebrew files were not deleted:
/usr/local/.com.apple.installer.keep
/usr/local/Frameworks/
/usr/local/Homebrew/
/usr/local/aws-cli/
/usr/local/bin/
/usr/local/etc/
/usr/local/include/
/usr/local/lib/
/usr/local/opt/
/usr/local/sbin/
/usr/local/share/
/usr/local/var/
You may wish to remove them yourself.


~$ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
==> Checking for `sudo` access (which may request your password)...
Password:
==> This script will install:
/usr/local/bin/brew
/usr/local/share/doc/homebrew
/usr/local/share/man/man1/brew.1
/usr/local/share/zsh/site-functions/_brew
/usr/local/etc/bash_completion.d/brew
/usr/local/Homebrew
==> The following new directories will be created:
/usr/local/Cellar
/usr/local/Caskroom

Press RETURN/ENTER to continue or any other key to abort:
==> /usr/bin/sudo /bin/mkdir -p /usr/local/Cellar /usr/local/Caskroom
==> /usr/bin/sudo /bin/chmod ug=rwx /usr/local/Cellar /usr/local/Caskroom
==> /usr/bin/sudo /usr/sbin/chown thn929 /usr/local/Cellar /usr/local/Caskroom
==> /usr/bin/sudo /usr/bin/chgrp admin /usr/local/Cellar /usr/local/Caskroom
==> /usr/bin/sudo /usr/sbin/chown -R thn929:admin /usr/local/Homebrew
==> /usr/bin/sudo /bin/chmod g+rwx /Users/thn929/Library/Caches/Homebrew
==> /usr/bin/sudo /usr/sbin/chown -R thn929 /Users/thn929/Library/Caches/Homebrew
==> Downloading and installing Homebrew...
HEAD is now at 251a098fb Merge pull request #16256 from Bo98/tap-json-fix
/usr/local/Homebrew/Library/Homebrew/standalone/init.rb:26:in `<top (required)>': undefined method `parent' for #<Pathname:/usr/local/Homebrew/Library/Homebrew/standalone> (NoMethodError)
	from /usr/local/Homebrew/Library/Homebrew/startup.rb:6:in `require_relative'
	from /usr/local/Homebrew/Library/Homebrew/startup.rb:6:in `<top (required)>'
	from /usr/local/Homebrew/Library/Homebrew/global.rb:4:in `require_relative'
	from /usr/local/Homebrew/Library/Homebrew/global.rb:4:in `<top (required)>'
	from /usr/local/Homebrew/Library/Homebrew/brew.rb:18:in `require_relative'
	from /usr/local/Homebrew/Library/Homebrew/brew.rb:18:in `<main>'
Failed during: /usr/local/bin/brew update --force --quiet
```

**Comment by thn929** [comment, marked as answer]
I fixed it.

### TL;DR 
After running the uninstall script, please go ahead and `rm -rf /usr/local/Homebrew/` to avoid running into conflicting Ruby versions.

### Details
I narrowed this down by dropping some `puts` in the `init.rb` and found that `LOAD_PATH` differs depending on how `init.rb` is invoked.

**A.** running a raw `ruby init.rb`, `LOAD_PATH` contains
```
/Library/Ruby/Gems/2.6.0/gems/did_you_mean-1.3.0/lib
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/vendor_ruby/2.6.0
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/vendor_ruby/2.6.0/x86_64-darwin21
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/vendor_ruby/2.6.0/universal-darwin21
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/vendor_ruby
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/x86_64-darwin21
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/universal-darwin21
```

**B.** running `brew` (which calls `init.rb`), `LOAD_PATH` contains
```
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/lib/ruby/vendor_ruby/2.6.0
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/lib/ruby/vendor_ruby/2.6.0/x86_64-darwin15
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/lib/ruby/vendor_ruby
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/lib/ruby/2.6.0
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/lib/ruby/2.6.0/x86_64-darwin15
```

Apparently portable-ruby 2.6.10_1 in the remnant `/usr/local/Homebrew/` had issues with `Pathname(...).parent`.

With me knowing very little about the internals of Homebrew, I did not clean up `/usr/local/Homebrew/` before attempting a reinstall because based on the language used in the uninstall script, it seemed like this was an optional step, therefore inconsequential to the reinstall. Well, here you have it, definitely not inconsequential. 

Perhaps a bug or an enhancement is in store. I'll leave this to someone more familiar with Homebrew.
